src/nix-env/user-env.cc

```diff
@@ -106,7 +107,9 @@ bool createUserEnv(EvalState & state, DrvInfos & elems,
        environment. */
     auto manifestFile = ({
         std::ostringstream str;
-        manifest.print(state.symbols, str, true);
+        manifest.print(state, str, PrintOptions {
+            .trackRepeated = false
+        });
```

Comment:
I think it'd be a safer choice to use the legacy format here instead. That will be one less worry when updating the new human-friendly format.